server/service/appconfig.go

```diff
@@ -270,15 +276,57 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p []byte) (*fleet.AppCo
 	// We apply the config that is incoming to the old one
 	appConfig.EnableStrictDecoding()
 	if err := json.Unmarshal(p, &appConfig); err != nil {
-		return nil, ctxerr.Wrap(ctx, &badRequestError{message: err.Error()})
+		return nil, ctxerr.Wrap(ctx, &fleet.BadRequestError{Message: err.Error()})
+	}
+	var legacyUsedWarning error
+	if appConfig.DidUnmarshalLegacySettings() {
```

Comment:
We should probably collect which ones are legacy and inform the user. That would save us a back and forth in support.

Not sure if we want this for this PR in particular, though.

Reply:
Yeah that would be a nice improvement. I think for now it's not too bad as there's only the `host_settings` key that indicates a legacy setting, but we should definitely add this to the next iteration of validation improvements. Depending on the discussion on the ticket regarding future iterations, I'll add this there.

Reply:
Added as part of https://github.com/fleetdm/fleet/issues/7835